android/src/main/java/com/mattermost/networkclient/Helper.kt

```diff
@@ -73,40 +76,42 @@ fun OkHttpClient.Builder.parseOptions(options: ReadableMap): OkHttpClient.Builde
     }
 
     // Retries
+    this.retryOnConnectionFailure(false);
     if (options.hasKey("retryPolicyConfiguration")) {
-        val retryPolicyConfiguration = options.getMap("retryPolicyConfiguration")!!
-        val retryType = retryPolicyConfiguration.getString("type")
-        val retryLimit = retryPolicyConfiguration.getDouble("retryLimit")
-        val retryExponentialBackoffBase = retryPolicyConfiguration.getDouble("exponentialBackoffBase")
-        val retryExponentialBackoffScale = retryPolicyConfiguration.getDouble("exponentialBackoffScale")
-        this.addNetworkInterceptor(RetryInterceptor(retryType, retryLimit.toInt(), retryExponentialBackoffBase, retryExponentialBackoffScale))
+        val retryPolicyConfiguration = options.getMap("retryPolicyConfiguration")!!.toHashMap();
+
+        val retryType = retryPolicyConfiguration["type"] as String?
+        val retryLimit = retryPolicyConfiguration["retryLimit"] as Double?
+        val retryExponentialBackoffBase = retryPolicyConfiguration["exponentialBackoffBase"] as Double?
+        val retryExponentialBackoffScale = retryPolicyConfiguration["exponentialBackoffScale"] as Double?
+        this.addInterceptor(RetryInterceptor(retryType, retryLimit?.toInt(), retryExponentialBackoffBase, retryExponentialBackoffScale))
     }
 
     // Headers
-    if(options.hasKey("headers")){
+    if (options.hasKey("headers")) {
         this.addNetworkInterceptor(HeadersInterceptor(options.getMap("headers")!!));
```

Comment:
`addNetworkInteceptor` can't call proceed more than once, because they act on the intermediately requests/responses; so they kick in before/after things like redirects - whereas `addInterceptor` doesn't care about those, plus a few more things details here:

https://square.github.io/okhttp/interceptors/#choosing-between-application-and-network-interceptors